### PR TITLE
Implement CDC integration tests and acceptance tests

### PR DESCRIPTION
**Pull request by alyakimenko**
### Description

- Implement CDC integration tests
- Implement acceptance test

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-s3/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: source/source.go [hunk 1]
```diff
@@ -102,6 +102,12 @@ func (s *Source) Read(ctx context.Context) (sdk.Record, error) {
 	return record, nil
 }
 
+// Ack does nothing. We don't need acks for the Snapshot or CDC iterators.
+// It just returns nil here in order to pass the acceptance tests properly.
+func (s *Source) Ack(ctx context.Context, position sdk.Position) error {
+	return nil
```
Comment (maksenius) [inline source/source.go:108]: please change it to
```
	sdk.Logger(ctx).Debug().Str("position", string(rp)).Msg("got ack")

	return nil
```

File: source/source_integration_test.go [hunk 2]
```diff
@@ -211,6 +216,95 @@ func TestSource_Snapshot_Continue(t *testing.T) {
...
+		is.NoErr(err)
+	})
+
+	s := new(Source)
+
+	err = s.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	// wait a bit for Vitess to get the "create table" query results from each shard
+	time.Sleep(time.Second * 3)
```
Comment (maksenius) [inline source/source_integration_test.go:247]: is it 3 second enough?
What will be if it takes tome more than 3 second?

Is it possible to do it without timeout?

File: source/source_integration_test.go [hunk 3]
```diff
@@ -211,6 +216,95 @@ func TestSource_Snapshot_Continue(t *testing.T) {
...
+	// switch to CDC iterator
+	_, err = s.Read(ctx)
+	is.Equal(err, sdk.ErrBackoffRetry)
+
+	err = insertRow(ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName, 1, "Bob")
+	is.NoErr(err)
+
+	time.Sleep(time.Second * 5)
+
+	readCtx, cancel := context.WithTimeout(ctx, time.Second*15)
```
Comment (maksenius) [inline source/source_integration_test.go:262]: Could you please tell me why we use context with timeout?

File: source/source_integration_test.go [hunk 4]
```diff
@@ -211,6 +216,95 @@ func TestSource_Snapshot_Continue(t *testing.T) {
...
+	defer cancel()
+
+	record, err := readWithRetry(readCtx, t, s, time.Second*5)
```
Comment (maksenius) [inline source/source_integration_test.go:265]: Is it possible run read function without retry?

File: source/source_integration_test.go [hunk 5]
```diff
@@ -211,6 +216,95 @@ func TestSource_Snapshot_Continue(t *testing.T) {
...
+	record, err = readWithRetry(readCtx, t, s, time.Second*5)
+	is.NoErr(err)
+	is.Equal(record.Metadata["action"], "insert")
+	is.Equal(record.Key.Bytes(), []byte(`{"int_column":2}`))
+	is.Equal(record.Payload.Bytes(), []byte(`{"int_column":2,"text_column":"Ben"}`))
+
+	err = s.Teardown(ctx)
+	is.NoErr(err)
+}
+
```
Comment (maksenius) [inline source/source_integration_test.go:307]: could you please add test for cdc from empty table

File: acceptance_test.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+	sdk.AcceptanceTest(t, &driver{
+		ConfigurableAcceptanceTestDriver: sdk.ConfigurableAcceptanceTestDriver{
+			Config: sdk.ConfigurableAcceptanceTestDriverConfig{
+				Connector:         Connector,
+				SourceConfig:      cfg,
+				DestinationConfig: cfg,
+				BeforeTest:        beforeTest(t, cfg),
+				GoleakOptions: []goleak.Option{
+					// these leaks spawn Vitess libraries, there's no way to stop them manually
```
Comment (maksenius) [inline acceptance_test.go:86]: could you please create issue for vittes library. It is only if you want, you can close this comment

File: source/source_integration_test.go (hunk 4 above)
Comment (alyakimenko) [reply source/source_integration_test.go:265]: Yeah, but we can get `ErrBackoffRetry`, that's why we need the retries

File: source/source.go (hunk 1 above)
Comment (alyakimenko) [reply source/source.go:110]: Done

File: source/source_integration_test.go (hunk 2 above)
Comment (alyakimenko) [reply source/source_integration_test.go:247]: If it takes more than 3 seconds we will get an error saying that a table doesn't exist. But in my experience, it never took longer than 3 seconds.

File: source/source_integration_test.go (hunk 5 above)
Comment (alyakimenko) [reply source/source_integration_test.go:307]: Added

File: source/source_integration_test.go (hunk 3 above)
Comment (alyakimenko) [reply source/source_integration_test.go:262]: If `readWithRetry` takes too long, we'll just exit the test with a fail

File: acceptance_test.go (hunk 6 above)
Comment (alyakimenko) [reply acceptance_test.go:86]: It will take some time to investigate from where exactly these problems come within the Vitess libraries, so I will close the comment for now

File: source/source_integration_test.go (hunk 4 above)
Comment (alyakimenko) [reply source/source_integration_test.go:265]: I mean data from Vitess may not come in real-time, and delays may occur

File: source/source_integration_test.go (hunk 2 above)
Comment (maksenius) [reply source/source_integration_test.go:247]: maybe we can check if table exist in loop with timeout. What do you think about it?

Comment (alyakimenko) [reply source/source_integration_test.go]: I've rewrote the `prepareData` function, now it creates test tables using `vtgateconn` and `session` and it seems that these things wait for a table to be created internally

**Review by maksenius — APPROVED** [review]
